server/middlewares/stats.js

```diff
@@ -31,7 +31,11 @@ const trackVisits = (req, res, next) => {
 }
 
 const trackTxSubmissionCount = mung.json((body, req, res) => {
-  if (req.originalUrl === '/api/txs/submit' && req.method === 'POST') {
+  if (
+    req.header('host') === 'https://adalite.io' &&
```

Comment:
I think you should check the 'origin' header

Reply:
And I would rather make a helper function `isSameOrigin(url1, url2)` or something like that, which would take the urls as strings, transform the to URL objects (https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) which does the parsing for you and you just compare the `origin` attributes of them which are stripped of any redundant slashes which may cause the comparison to return false negatives.

Reply:
I would also use some url normalization, to strip "www" from the compared urls: https://www.npmjs.com/package/normalize-url Feel free to add it as a server dependency

Reply:
Is it really needed? I only see the point if comparing URL's with params, but comparing url origins (after wrapping them in URL object) should be sufficient.

Reply:
I think so, the 'www' prefix would pass as a false negative, here is a code sample showing what I mean:
```
new URL('https://www.adalite.io').origin
-> "https://www.adalite.io"
new URL('https://adalite.io').origin
-> "https://adalite.io"
```

Reply:
I see.